### PR TITLE
update cholesky in UKI

### PR DESCRIPTION
**Pull request by Zhengyu-Huang**
Update the Cholesky factor computation in UKI

When the matrix is not positive definite, compute SVD of the matrix, replace 0 singular values as the smallest nonzero singular value (with a threshold 1e-8), and then compute the Cholesky factor.

### CODE REVIEW

File: src/UnscentedKalmanInversion.jl [hunk 1]
```diff
@@ -172,7 +172,16 @@ function construct_sigma_ensemble(
 
     c_weights = process.c_weights
 
-    chol_xx_cov = cholesky(Hermitian(x_cov)).L
+    # compute cholesky factor L of x_cov
+    try
+        global chol_xx_cov = cholesky(Hermitian(x_cov)).L
```
Comment (ilopezgp) [inline src/UnscentedKalmanInversion.jl:177]: Is it safe to use `global` here @charleskawczynski ?

**Comment by ilopezgp** [comment]
Can you pass a unit test of `construct_sigma_ensemble` passing a positive semidefinite covariance matrix?

File: src/UnscentedKalmanInversion.jl (hunk 1 above)
Comment (Zhengyu-Huang) [reply src/UnscentedKalmanInversion.jl:177]: I will add a unit test, if I do not put global here, I get an error. the scope of try and catch is weird

Comment (charleskawczynski) [inline src/UnscentedKalmanInversion.jl:177]:
```suggestion
    local chol_xx_cov
    try
        chol_xx_cov = cholesky(Hermitian(x_cov)).L
```

File: src/UnscentedKalmanInversion.jl [hunk 2]
```diff
@@ -172,7 +172,16 @@ function construct_sigma_ensemble(
...
+    catch
+        _, S, Ut = svd(x_cov)
+        # find the first singular value that is smaller than 1e-8
+        ind_0 = searchsortedfirst(S, 1e-8, rev = true)
+        S[ind_0:end] .= S[ind_0 - 1]
+        global chol_xx_cov = (qr(sqrt.(S) .* Ut).R)'
```
Comment (charleskawczynski) [inline src/UnscentedKalmanInversion.jl:183]:
```suggestion
       chol_xx_cov = (qr(sqrt.(S) .* Ut).R)'
```

File: src/UnscentedKalmanInversion.jl (hunk 1 above)
Comment (charleskawczynski) [reply src/UnscentedKalmanInversion.jl:177]: I would typically use `local` here, maybe global works the same way? I think `local` is more intuitive.

Comment (Zhengyu-Huang) [reply src/UnscentedKalmanInversion.jl]: just update to `local` and add a test

**Review by ilopezgp — APPROVED** [review]
LGTM, thank you @Zhengyu-Huang !

**Comment by ilopezgp** [comment]
Closes #126

**Pull request merged**